### ISSUE TITLE
[bug] [alpha.17] [windows] Cannot instantiate more than one window on some machines

### ISSUE THREAD

**Issue by kris-ava** [tauri-apps/tauri#8196]
### Describe the bug

When I try to instantiate more than one window - either using `WindowBuilder` or by having more than one entry in the `windows` config array - on some Windows machine the app consistently dies and shows `failed to create webview: WebView2 error: WindowsError(Error { code: HRESULT(0x8007139F), message: "The group or resource is not in the correct state to perform the requested operation." })` on the console.

This is only reproducible on some Windows machines (I managed to reproduce it on 1 out of 3 I tried), but on that one it is produced very consistently. I suspect it's linked to OneDrive (the machine where it fails is using OneDrive).

The workaround was to remove all the entries from the `windows` config array, and instantiate all the windows manually while setting different `data_directory` built from `app_data_dir` with different subdirectories attached to it. I suspect it might be a good idea to instantiate all windows with their own data_directory by default, at least on Windows.

### Reproduction

 * Run npm create tauri-app@latest --alpha
 * Add a second entry to the `windows` array in the config
 * Run the app

### Expected behavior

Two windows open

### Platform and versions

```text
[✔] Environment
    - OS: Windows 10.0.20348 X64
    ✔ WebView2: 118.0.2088.76
    ✔ MSVC: Visual Studio Community 2022
    ✔ rustc: 1.72.0 (5680fa18f 2023-08-23)
    ✔ cargo: 1.72.0 (103a7ff2e 2023-08-15)
    ✔ rustup: 1.26.0 (5af9b9484 2023-04-05)
    ✔ Rust toolchain: stable-x86_64-pc-windows-msvc (default)
    - node: 18.13.0
    - yarn: 1.22.19
    - npm: 8.19.3

[-] Packages
    - tauri [RUST]: 2.0.0-alpha.17
    - tauri-build [RUST]: 2.0.0-alpha.11
    - wry [RUST]: 0.34.2
    - tao [RUST]: 0.23.0
    - @tauri-apps/api [NPM]: 2.0.0-alpha.11
    - @tauri-apps/cli [NPM]: 2.0.0-alpha.17

[-] App
    - build-type: bundle
    - CSP: unset
    - distDir: ../dist
    - devPath: http://localhost:1420/
    - bundler: Vite
```

### Stack trace

```text
failed to create webview: WebView2 error: WindowsError(Error { code: HRESULT(0x8007139F), message: "The group or resource is not in the correct state to perform the requested operation." })
```

**Comment by DexterHill0** [comment]
I can also reproduce this issue with the lastest alpha releases, while having 2 windows defined in the Tauri config. No matter how many times I run it no window(s) will ever be visibile, but only some of the time is that error explicitly printed.
This seems like a common problem as it was reported in issues #8143, #7897, #7826 and #8196 as well.

**Comment by thewh1teagle** [comment]
I experience this bug too on the beta release of Tauri.
It happens to me when I drag the window before it fully loaded (on dev mode), so I guess it's some race condition with Window postition and the initial logic.

**Comment by ImageDeeply** [comment]
@FabianLars Any thoughts on timing? We switched from 1.x to 2.x for some useful Mac improvements, but are now blocked on Windows 11 on 3 different machines. It's alpha so at our own risk; that's fine. Just trying to decide whether we can afford to wait or have to go back to 1.x.

**Comment by FabianLars** [comment]
No idea. There are 2 bugs here though. one in webview2 itself (fixed in their canary builds) which makes it crash basically always, and something in tauri that crashes it _sometimes_ (this is what this issue was about).

I hope microsoft pushes that fix to stable asap. We're close to their regular release date though so hopefully it'll be part of that.

For the tauri specific issue i'm not sure how to best handle the race condition yet (using seperate data dirs as proposed in OP's message is not a good solution imo, since those folders are already larger than we'd like), we're looking into it though.

**Comment by FabianLars** [comment]
Alrighty, so i spent the whole day on this and did not get much further.

All i know is that the crash happens here https://github.com/tauri-apps/wry/blob/dev/src/webview2/mod.rs#L304-L316. I tried to get more info with debuggers but just couldn't get more than the error we already had...

Also i think the issue is that _something_ goes wrong with the first webview data dir registration which makes all subsequent ones fail because i tried to add some auto retry mechanism by either just retrying the linked call or including a new webview2 environment object as suggested in their docs. Both did not work, they just kept on failing with the same error.

If nobody in the team (or the community) has any more ideas i will open an issue in the webview2 repo over the weekend because i consider this a blocker for v2.

p.s. to reproduce this, just use the multiwebview example in wry, you may have to set the same user data dir for all 4 webviews but letting the webview use the default path also causes this error, just a bit less frequent i think. You can also reproduce this with the multiwindow example if you spam the button **really** fast, or easier by changing the code to spawn 2-4 windows in quick succession.

p.p.s. Until the next webview2 runtime release, don't forget to use the canary runtime to test this.

**Comment by ImageDeeply** [comment]
Thanks for digging in.

**Comment by thewh1teagle** [comment]
Good news!
I think Microsoft finally fixed the critical bug they had and released it into stable version!

To update it:
1. Open microsoft edge (stable, the regular) from search
2. Open `edge://settings/help` and check that version is ` 120.0.2210.144` or greater or wait for it to update
3. Rerun your app and ensure that it uses version ` 120.0.2210.144` or later using `tauri::webview_version()`

**Comment by amrbashir** [comment]
Nice, just tried and couldn't reproduce any crashes as before, will close this for now but we will keep monitoring the situation. If it crashes for anyone again, let us know.

**Comment by Borber** [comment]
I also had the same problemon all the latest dependencies, and I installed only one webview2 runtime 

![image](https://github.com/tauri-apps/tauri/assets/30563826/9acbcd5b-aa29-41c4-ae08-a7a02aae2689)

Deleting the configuration list as you mentioned and manually creating all windows temporarily solved this problem. 

**Thank you very much**

OS: Windows 10 LSTC 21H2 19044.3930 Windows Feature Experience Pack 1000.19053.1000.0

```toml
[build-dependencies]
tauri-build = { version = "2.0.0-alpha.14", features = [] }

[dependencies]
tauri = { version = "2.0.0-alpha.21", features = ["tray-icon"] }
tauri-plugin-process = "2.0.0-alpha.6"
tauri-plugin-single-instance = "2.0.0-alpha.6"

```


```json
"dependencies": {
    "@solidjs/router": "^0.10.10",
    "@tauri-apps/api": "^2.0.0-alpha.13",
    "@tauri-apps/plugin-shell": "^2.0.0-alpha.3",
    "@tauri-apps/plugin-process": "^2.0.0-alpha.5",
    "solid-js": "^1.8.12"
  },
  "devDependencies": {
    "@tauri-apps/cli": "2.0.0-alpha.21",
    "@types/node": "^20.11.10",
    "@typescript-eslint/eslint-plugin": "^6.20.0",
    "@typescript-eslint/parser": "^6.20.0",
    "eslint": "^8.56.0",
    "eslint-plugin-simple-import-sort": "^10.0.0",
    "eslint-plugin-solid": "^0.13.1",
    "typescript": "^5.3.3",
    "vite": "^5.0.12",
    "vite-plugin-solid": "^2.9.1"
  }
```

**Comment by gulbanana** [comment]
I am still experiencing a crash when declaring multiple windows in tauri.conf.json. I have WebView2 121.0.2277.106. The error is "The group or resource is not in the correct state to perform the requested operation.".

**Comment by Borber** [comment]
> I am still experiencing a crash when declaring multiple windows in tauri.conf.json. I have WebView2 121.0.2277.106. The error is "The group or resource is not in the correct state to perform the requested operation.".

I later chose to create it manually, and later, I directly adjusted it to a single-window application.

**Comment by thewh1teagle** [comment]
I experience this again.
Webview2 `Version 121.0.2277.128 (Official build) (64-bit)`
Windows 11
Latest wry [#49d2](8c86fbaf51cd970737cc070583318d4b532349d2)
Multiwebview wry example

Reproduce:
```console
git clone https://github.com/tauri-apps/wry
cd wry
cargo run --example multiwebview
```
wry issue: https://github.com/tauri-apps/wry/issues/1173

**Comment by mrdkprj** [comment]
The argument passed to ICoreWebView2EnvironmentOptions.SetAdditionalBrowserArguments seems to be inconsistent in [wry](https://github.com/tauri-apps/wry/blob/f8761fbaaa38ef434740e371041ea099fcaf6037/src/webview2/mod.rs#L201).
Default argument string is "--disable-features=msWebOOUI,msPdfOOUI,msSmartScreenProtection --autoplay-policy=no-user-gesture-required" and its code points should be

` [45, 45, 100, 105, 115, 97, 98, 108, 101, 45, 102, 101, 97, 116, 117, 114, 101, 115, 61, 109, 115, 87, 101, 98, 79, 79, 85, 73, 44, 109, 115, 80, 100, 102, 79, 79, 85, 73, 44, 109, 115, 83, 109, 97, 114, 116, 83, 99, 114, 101, 101, 110, 80, 114, 111, 116, 101, 99, 116, 105, 111, 110, 32, 45, 45, 97, 117, 116, 111, 112, 108, 97, 121, 45, 112, 111, 108, 105, 99, 121, 61, 110, 111, 45, 117, 115, 101, 114, 45, 103, 101, 115, 116, 117, 114, 101, 45, 114, 101, 113, 117, 105, 114, 101, 100]`

But code points sometimes become different like [16560, 21260, 607], [26016, 48040, 354].

Error occurs when first WebView is created with invalid argument and the second WebView is created with valid argument.
Webviews are created successfully when both arguments are valid or invalid(ignorable).

I don't even know if this inconsistency is expected and irrelevant to the error, but this was the only difference I observed.

**Comment by amrbashir** [comment]
thank you @mrdkprj, that was indeed the problem. I am in the process of refactoring our code in WRY which touched this area of code and I didn't see this error in a while. Good to finally know what exactly caused the issue. 

The refactor is being done on https://github.com/tauri-apps/wry/tree/refactor-backends if anyone wants to test it. Also subscribe to https://github.com/tauri-apps/wry/issues/1173 to get notified when refactor is done

**Comment by datner** [comment]
I am still experiencing this 😭 
latest tauri latest rust latest webview2 latest everything

**Comment by Elendiar** [comment]
Rust version: `Ok("129.0.2792.52")` Same happens with me 
`[2024-09-26][05:38:59][tauri_runtime_wry][ERROR] failed to create webview: WebView2 error: WindowsError(Error { code: HRESULT(0x8007139F), message: "The group or resource is not in the correct state to perform the requested operation." })`
My program is appbar with buttons, when user press button for internal web resource, js create `new WebviewWindow` for this entity with this error, and window flashes on create/destroy.

Moreover, I have been developing it for a couple of months (or rather, migrate from electron node ffi to tauri windows-rs), and this part of the code was made a long time ago, but now it suddenly stopped working. The program also creates some windows from rust backend - but there are no problems with them. Only with windows from the frontend.

UPD: This happened after i add `additional_browser_args()` to main window builder. Even it is empty string `""`, or `"--enable-features=msWebView2EnableDraggableRegions --disable-features=ElasticOverscroll"` - failed to create webview occures!

**Comment by eythaann** [comment]
> Rust version: `Ok("129.0.2792.52")` Same happens with me `[2024-09-26][05:38:59][tauri_runtime_wry][ERROR] failed to create webview: WebView2 error: WindowsError(Error { code: HRESULT(0x8007139F), message: "The group or resource is not in the correct state to perform the requested operation." })` My program is appbar with buttons, when user press button for internal web resource, js create `new WebviewWindow` for this entity with this error, and window flashes on create/destroy.
> 
> Moreover, I have been developing it for a couple of months (or rather, migrate from electron node ffi to tauri windows-rs), and this part of the code was made a long time ago, but now it suddenly stopped working. The program also creates some windows from rust backend - but there are no problems with them. Only with windows from the frontend.
> 
> UPD: This happened after i add `additional_browser_args()` to main window builder. Even it is empty string `""`, or `"--enable-features=msWebView2EnableDraggableRegions --disable-features=ElasticOverscroll"` - failed to create webview occures!

x2 when I try to add additional_browser_args the app start crashing with the same error. if you open or find a issue/ticket for this tag me please.

**Comment by Elendiar** [comment]
> if you open or find a issue/ticket for this tag me please.

I already created [issue](https://github.com/tauri-apps/tauri/issues/11144), and it is normal, but not documented behaviour. Specifying `data_directory` for builder with `additional_browser_args` fix this.

**Comment by eythaann** [comment]
> > if you open or find a issue/ticket for this tag me please.
> 
> I already created [issue](https://github.com/tauri-apps/tauri/issues/11144), and it is normal, but not documented behaviour. Specifying `data_directory` for builder with `additional_browser_args` fix this. 

hmmm I see, this should be documented cuz it is not an intuitive thing. btw thanks for the information! :D

**Comment by apalacean-panaya** [comment]
Hi, we have the same issue when we want to open another window:

'Runtime(CreateWebview(WebView2Error(WindowsError(Error { code: 0x8007139F, message: The group or resource is not in the correct state to perform the requested operation. }))))'

WebView version is 136.0.3240.92.
How could we fix it?

**Comment by FabianLars** [comment]
sounds like the same issue like above. specifying the same additionalBrowserArgs for all windows, or setting a data_directory for windows that don't use the default args should fix that.

if that doesn't help, please share the output of the `tauri info` command for a start (if you prefer, a new issue is fine too)
